src/garage/torch/distributions/tanh_normal.py

```diff
@@ -0,0 +1,117 @@
+                                 computed.
+            pre_tanh_value(torch.tensor): The value prior to having the tanh
+                                          function applied to it but after it
+                                          has been sampled from the normal
+                                          distribution.
+
+        Returns:
+            torch.tensor with log likelihood.
+
+        Notes:
+            - pre_tanh_value should usually not be None, but as the option
+            available just in case a value is not/cannot be passed.
+
+        """
+
+        def _clip_but_pass_gradient(x, lower=-1., upper=1.):
+            clip_up = (x > upper).float()
+            clip_low = (x < lower).float()
+            with torch.no_grad():
+                clip = ((upper - x) * clip_up + (lower - x) * clip_low)
+            return x + clip
+
+        if pre_tanh_value is None:
+            pre_tanh_value = torch.log((1 + value) / (1 - value)) / 2
+        ret = self._normal.log_prob(pre_tanh_value) - torch.sum(torch.log(
+            _clip_but_pass_gradient((1. - value**2), lower=0., upper=1.) +
+            self._epsilon),
+                                                                axis=-1)
+        return ret
+
+    def sample(self, return_pre_tanh_value=False):
+        """Return a sample, sampled from this Tanh Normal Distribution.
+
+        Gradients should *not* pass through this operation.
+
+        Args:
+            return_pre_tanh_value(bool): When true, returns the sampled value
+                                         before the Tanh transform is applied.
+
+        Returns:
+            named tuple with fields "pre_tanh_action" and "action"
+
+        """
+        action_infos = namedtuple('action_infos',
+                                  ['pre_tanh_action', 'action'])
+        with torch.no_grad():
+            z = self._normal.sample()
+            if return_pre_tanh_value:
+                return action_infos(z, torch.tanh(z))
+            return action_infos(None, torch.tanh(z))
+
+    def rsample(self, return_pre_tanh_value=False):
+        """Return a sample, sampled from this Tanh Normal Distribution.
+
+        Gradients should pass through this operation.
+
+        Args:
+            return_pre_tanh_value(bool): When true, returns the sampled value
+                                         before the Tanh transform is applied.
+
+        Returns:
+            named tuple with fields "pre_tanh_action" and "action"
+
+        """
+        action_infos = namedtuple('action_infos',
+                                  ['pre_tanh_action', 'action'])
+        z = self._normal.rsample()
+        if return_pre_tanh_value:
+            return action_infos(z, torch.tanh(z))
+        return action_infos(None, torch.tanh(z))
+
+    @property
+    def loc(self):  # noqa D102
+        return self._normal.loc
+
+    @property
+    def mean(self):  # noqa D102
```

Comment:
Might want to note in a docstring that this value is technically not correct (but much easier to compute than the true value, and typically quite close).